### PR TITLE
Search Block 3: Add Basic UI

### PR DESCRIPTION
**Pull request by AmandaRiu**
Related to #3073. Updates the gutenberg reference to point to the branch that adds the Search block basic UI (dev mode only). This is just one of many PR steps for this block. 

### Gutenberg PR
https://github.com/WordPress/gutenberg/pull/29045

### To test
Please see the Gutenberg PR for testing instructions. 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Review by cameronvoell — APPROVED** [review]
Good to merge once we update the ref after gutenberg side is merged 👍

**Comment by AmandaRiu** [comment]
Closing as the current gutenberg reference already includes these changes.

**Pull request closed without merging**
